src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -7468,6 +7469,103 @@ namespace WaterThermalTanks {
 		if ( HeatPump.Mode == HeatMode ) {
 			// HPWH was heating last iteration and will continue to heat until the set point is reached
 			HPPartLoadRatio = 1.0;
+			if ( TankTemp > SetPointTemp ) { // tank set point temp may have been reduced since last iteration and float mode may be needed
+				HeatPump.Mode = FloatMode;
```

Comment:
Note all this logic was added to HeatMode (whereas before only PLR=1 was used) to account for times when either the SP was reduced or the use side nodes could meet the set point. This corrects the defect file.